tcms/testruns/tests/test_views.py

```diff
@@ -454,6 +456,18 @@ def test_update_selected_case_runs(self):
         self.assertContains(response, _('%d CaseRun(s) updated:') % 1)
         self.assertEqual(self.case_run_1.case.text, "Scenario Version 2")
 
+    def test_update_selected_case_runs_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+
+        self.assertNotEqual(self.case_run_1.case.history.latest().history_id,
+                            self.case_run_1.case_text_version)
+        response = self.client.post(self.update_url,
+                                    {'case_run': [self.case_run_1.pk]},
+                                    follow=True)
+
+        self.assertNotContains(response, _('%d CaseRun(s) updated:') % 1)
+        self.assertEqual(self.case_run_1.case.text, "Scenario Version 2")
```

Comment:
This is wrong. It should be `NotEqual` and it should be failing right now, i.e. you are not able to update the text.

In reality see how the above test should work and change this one accordingly. We can't compare the objects directly b/c that will resolve through the foreign keys. Instead use `get_text_with_version()` and assert NotEqual, also make sure `case_text_version` is still different from the latest history record (refresh_from_db is a must here).